src/test/run-pass/panic-runtime/abort.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// compile-flags:-C panic=abort
+// no-prefer-dynamic
+
+use std::process::Command;
+use std::env;
+
+struct Bomb;
+
+impl Drop for Bomb {
+    fn drop(&mut self) {
+        std::process::exit(0);
+    }
+}
+
+fn main() {
+    let mut args = env::args_os();
+    let me = args.next().unwrap();
+
+    if let Some(s) = args.next() {
+        if &*s == "foo" {
+
+            let _bomb = Bomb;
+
+            panic!("try to catch me");
+        }
+    }
+    let s = Command::new(env::args_os().next().unwrap()).arg("foo").status();
```

Comment:
Ah I just find myself very commonly writing tests which failure or success involves modification or termination of the entire process in some flavorful fashion. For example this tests "succeeds" by correctly aborting rather than running destructors. Other tests like those for `CommandExt::exec` succeed by taking over the entire process.

This method of exec-yourself-with-some-other-arguments basically provides a very nice, cross platform, and reliable sandbox for spawning a fresh new process with your code running. That way the actual test itself can happen in the subprocess and the parent can then verify the results.